Restructure use case section and add more content on auctions

nebojsa-io opened the pull request:
Contribute to #710 

Changed the "Use Cases" section of the website.

* It now includes 3 main sections:
  - Payments: a stub with an initial intro to hydra-pay (to be expanded with more use cases later)
    + This category now holds the two legacy use case pages about inter-wallet payments and pay-per-use apis
  - Auctions: includes the description of the current implementation done with mlabs + descriptions of extension use cases that can be built on top
  - Other: Now only holds the "poker game" use case

* The legacy NFT Auction article was removed

* Added redirects to not break links from external

---

<!-- Consider each and tick it off one way or the other -->
* [x] CHANGELOG updated or not needed
* [x] Documentation updated or not needed
* [x] Haddocks updated or not needed
* [x] No new TODOs introduced or explained herafter

nebojsa-io committed:
First test with use case page

Initial test

nebojsa-io committed:
First test with use case page

Initial test

nebojsa-io committed:
Updated auctions section in use-cases

nebojsa-io committed:
Updated auctions section in use-cases

nebojsa-io committed:
Restructured "use cases" section. There are now 3 main sections: payments, auctions, and example use cases

nebojsa-io committed:
Restructured "use cases" section. There are now 3 main sections: payments, auctions, and example use cases

nebojsa-io commented:
Forgot to add: there are still a few missing diagrams, with placeholders on the top page for auctions use cases. These placeholders can/should be removed since I don't think the diagrams are necessary. The text is self-explanatory even without them.

pgrange commented on docs/use-cases/hydra-for-payments/index.md:
```diff
@@ -0,0 +1,35 @@
+# Hydra for Auctions
```
```suggestion
# Hydra for Payment
```

pgrange commented on docs/use-cases/nft-auction/index.md:
```diff
@@ -1,28 +1,37 @@
+> Using Hydra for Digital Asset Auctions
 
-NFT drops can be a real problem when run on L1, generating a lot of UTxO and transactions in a short amount of time, clogging the network and leading to congestion and general slowdowns. The problem seems to come not much from the NFTs themselves but from the large of number of bidding transactions people post to grab some.
+### Introduction: Using Hydra for Digital Asset Auctions 
```
```suggestion
### Using Hydra for Digital Asset Auctions 
```

pgrange commented on docs/use-cases/nft-auction/index.md:
```diff
@@ -1,28 +1,37 @@
-NFT drops can be a real problem when run on L1, generating a lot of UTxO and transactions in a short amount of time, clogging the network and leading to congestion and general slowdowns. The problem seems to come not much from the NFTs themselves but from the large of number of bidding transactions people post to grab some.
+### Introduction: Using Hydra for Digital Asset Auctions 
+Cardano makes minting and sending non-fungible tokens (NFTs) cheap and easy, as the accounting system for non-ADA tokens is hosted in the ledger itself (alongside ADA) instead of complicated and bug-prone custom smart contracts. This has led to a vibrant NFT ecosystem on Cardano for art, music, identity, real estate, gaming, service subscriptions, etc.
+[Diagram showing all the icons for NFT projects]
```
I can't see no diagram.

nebojsa-io replied:
removing...

pgrange commented on docs/use-cases/nft-auction/index.md:
```diff
@@ -1,28 +1,37 @@
+High-quality marketplaces now exist on Cardano to list, view, and purchase all sorts of NFTs. These NFT marketplaces have very friendly user interfaces that can neatly display images/animations, rarity charts, royalty terms, and other metadata for the NFTs, which can be purchased at the seller’s list price or via a buyer’s alternative offer. However, the novelty of the digital and non-digital ass ...
 
-We sketch a way to run NFT auctions inside Hydra Head that would alleviate that problem. Here is an outline of the sequence of transactions involved, both on-chain and off-chain:
+:::tip [Diagram of UI of a typical marketplace on Cardano]
```
I can't see no diagram

nebojsa-io replied:
removing...

pgrange commented on docs/use-cases/nft-auction/index.md:
```diff
@@ -1,28 +1,37 @@
-- The auctioneer posts a settlement transaction that consumes all current bids at some point, producing a voucher for the NFT to be sent to Bob if he pays V2'. The voucher is only valid if produced by this script, and there might be some reserve price to ensure price does not fall below some threshold;
-- Then Head is closed and the voucher is posted on-chain;
-- Bob can redeem his NFT, paying V2' to the auctioneer.
+:::tip [Diagram for how an auction works vs marketplace]
```
I can't see no diagram

nebojsa-io replied:
removing...

pgrange commented on docs/use-cases/nft-auction/index.md:
```diff
@@ -1,28 +1,37 @@
+Auctions should be lively and efficient – bidders enter auctions (often serendipitously) with limited time and attention spans, and they will leave if it takes too long for them to make their bids or observe other bidders’ bids. This bidder experience is hard to implement directly on Cardano’s main network (L1), where L1 transactions take time to be added to a block (~10–60 seconds) and confirmed ...
 
-The bidders run the same risk of bidding in a rigged auction but in the worst case, they can refuse to pay for the NFT. Anyhow, this setup would offer already a much better security story than all the fully custodial NFT drops done on Cardano today
+:::tip [Diagram for L1 transaction delays]
```
I can't see no diagram

nebojsa-io replied:
removing...

pgrange commented on docs/use-cases/nft-auction/index.md:
```diff
@@ -1,28 +1,37 @@
-:::
+One possible path to realizing this vision could progress along the following set of milestones that we believe are feasible to implement using Hydra and its anticipated improvements:
+
+- Delegated voucher auction (invitational) – the first complete prototype of an auction that can host its bidding process on L2.
```
I would make it more explicit that this prototype does, indeed, exist.

nebojsa-io replied:
done

pgrange commented on docs/use-cases/nft-auction/index.md:
```diff
@@ -1,28 +1,37 @@
+
+- Always-on delegated auction service (single-head) – This would allow a single persistent Hydra head to host multiple auctions over time without closing. Hydra Head operators would be able to build a business model offering L2-hosting services for auctions.
+
+- Always-on delegated auction service (multi-head) – This would allow an auction to split its bidding process among multiple Hydra Head hosts, reducing the reliance on any one Hydra Head.
```
_Hydra Head hosts_, in my mind, refers to the hosts that are hosting a Hydra node which is a party in a given head. So when I read this sentence I feel like we want to say that you could deploy your auction in several heads to not depend on one single host. But it's already the case with one single head: you do not depend on one single host but on all the hosts that participate in the head.

So I think I do not understand this sentence. I could use some more explanation here.

nebojsa-io replied:
I think it should say "among multiple Hydra heads". You would have some degree of centralization in a delegated model. So choosing between different "service providers" (i.e. different delegated heads) would be a step more decentralized than a single service provider (even if it is multiple node operators running a single head - since they will likely be not too many, they can collude to censor bids).

I'll change it to "among multiple Hydra heads".

pgrange reviewed (changes requested):
We can't merge without the diagrams, I think, because merging would publish on the website and it would look a bit broken without them, in my opinion.

nebojsa-io committed:
Update docs/use-cases/nft-auction/index.md

Co-authored-by: Pascal Grange <pgrange@users.noreply.github.com>

nebojsa-io committed:
Update docs/use-cases/hydra-for-payments/index.md

Co-authored-by: Pascal Grange <pgrange@users.noreply.github.com>

nebojsa-io committed:
Cleaned up and resolved feedback from #964

ch1bo commented on docs/use-cases/example-use-cases/_category_.json:
```diff
@@ -0,0 +1,4 @@
+{
+  "label": "Example Use Cases",
```
I would like to suggest naming this section something like "Other" use cases. Also, I would suggest not to re-use `-use-cases` in the directory name.

Should: rename directory and contents to `docs/use-cases/other/`

ch1bo commented on docs/use-cases/example-use-cases/archive/_category_.json:
Should: remove everything in archive

No need to keep it around

ch1bo commented on docs/use-cases/example-use-cases/inter-wallet-payments/_category_.json:
I think we can simplify this greatly by not using definite positions and we do not need this file as the first headline of `index.md` will be used (or customizable via the front-matter in the `inter-wallet-payments/index.md`

Should: remove this file
Could: rename `inter-wallet-payments/index.md` -> `inter-wallet-payments.md`

ch1bo replied:
Counter-argument against the renaming to `inter-wallet-payments.md` is to keep pictures and other assets local to the markdown like in the `poker-game/`

ch1bo commented on docs/use-cases/hydra-for-payments/_category_.json:
```diff
@@ -0,0 +1,4 @@
+{
+  "label": "Hydra for Payments",
```
Do we really want to repeat "Hydra for" in each use case section?

If we avoid that, we can easily keep the `inter-wallet-payments` and `pay-per-use-api` documents in the "Payments" category until we have something more structured.

Could: Rename to "Payments" and `use-cases/payments/` accordingly, and move the payment use case documents beneath this directory

ch1bo commented on docs/use-cases/hydra-for-payments/index.md:
```diff
@@ -0,0 +1,43 @@
+
+## Using Hydra for Peer-To-Peer Payments
+
+### Hydra for Payments Project
```
This is a very repetitive list of headlines. I would suggest to remove headlines of line 1 and 10 to make it look less repetetive

ch1bo commented on docs/use-cases/hydra-for-payments/index.md:
```diff
@@ -0,0 +1,43 @@
+# Hydra for Payments
+
+> Using Hydra for Peer-To-Peer Payments
```
These subtitles are a bit pointless as they just re-state the other headline(s).

Should: remove it

ch1bo commented on docs/use-cases/hydra-for-payments/index.md:
```diff
@@ -0,0 +1,43 @@
+
+> Using Hydra for Peer-To-Peer Payments
+
+:::warning This section is a stub 
```
Should: use `:::caution` to look less dangerous (yellow)

ch1bo replied:
As I also propose to move the legacy (but still nicely written sections) about payments into this category as well, we could give a quick disclaimer that these ideas still are somehow relevant and need to be adopted in a more consistent portfolio on payments.

ch1bo commented on docs/use-cases/hydra-for-payments/index.md:
```diff
@@ -0,0 +1,43 @@
+
+### An Example Configuration
+
+![](./hydra-for-payments.png)
```
This is a cool picture, but it looks horribly tilted. @chobbins @yasuke do you think we could get a more.. rectangular version of this visual? :)

ch1bo commented on docs/use-cases/nft-auction/always-on-service-single/index.md:
```diff
@@ -0,0 +1,22 @@
+# Always-On Delegated Auction Service (Single-Head)
```
This and other titles are a bit long. Can we find shorter ones?

ch1bo reviewed (commented):
I think this is a great start. I would like to discuss my ideas of categorizing things, shortening names and headlines.

ch1bo committed:
Remove archived use case pages

ch1bo committed:
Re-organized use-cases categories and slugs

ch1bo committed:
Make payments/auctions headlines consistent

ch1bo committed:
Fix links from topologies -> use cases

ch1bo commented on docs/use-cases/auctions/delegated-voucher-sdk/index.md:
```diff
@@ -0,0 +1,37 @@
+# SDK for Delegated Voucher Auctions
+
+> Modular SDK for App developers supporting delegated voucher auctions
```
An SDK is no use case?

I think this should be mentioned in the corresponding hydra-auction pages.

ch1bo replied:
@nebojsa-io and I discussed that we should try to incorporate this into the relevant sections of the invitational/open delegated auction use case instead.

ch1bo reviewed (commented):
Incorporated the changes I have suggested already. Besides the very verbose auction category and document titles, to be discussed:

ch1bo committed:
Add redirects to not break links from external

As github pages does not support server-side redirects (HTTP 301
responses), we use a docusaurus plugin to do generate pages with
client-side redirects.

ch1bo committed:
Update Hydra for payments overview diagram

ch1bo committed:
Move category files into file headers and updated auctions index

Move things more into actual page titles + subtexts and us the
DocCardList for the intro page.

ch1bo committed:
Re-order navbar items

ch1bo reviewed (approved):
Let's merge this now and see how it looks on the website. @nebojsa-io should come up with more ideas after that. If we like this approach, the payments vertical could also be treated the same.
